lib/akismet.js

```diff
@@ -37,16 +41,13 @@ var Akismet = function(options) {
       blog : this.blog
     })
     .set('User-Agent', this.userAgent)
-    .end(function(err, res) {
-      if (err) cb(err, null);
-      else if (res.text == 'valid') cb(null, true);
-      else if (res.header['x-akismet-debug-help']) 
-        cb(new Error(res.header['x-akismet-debug-help']), false);
-      else if (res.text == 'invalid')
-        cb(new Error('Invalid API key'), false);
-      else 
-        cb(new Error(res.text), false);
-    });
+    .endAsync()
+    .then(function(res) {
+      if (res.text == 'valid') return true;
+      if (res.text == 'invalid') return false;
+      throw new Error(res.text);
```

Comment:
@andapopovici This is for semantic reasons, I don't like not having that message but I couldn't find a better way to deal with it. In the callback implementation I was returning the `res.header` field as the error, and setting the result to be `false`. So I was returning a result and an error. Promises don't let you do that, things must either reject with an error and no result or they must succeed with no error and have a result. 

I figured that, in the scenario they have an invalid key and there is a debug message, rejecting the promise is not the "proper" thing to do since the check didn't _fail_, the result was just negative. But if I resolve the promise with the value `false` there is no way to also pass an error along. 

I'm open to any suggestions for better implementations here :) I was thinking about returning an object or an array instead, where the first parameter is the result and the second parameter is the debug message. That felt slightly hacky though.